hasura/Tiltfile

```diff
@@ -13,43 +13,40 @@ DEFAULT_HASURA_RESOURCE_NAME = 'hasura'
 def hasura_console(release_name='',
             path='.', 
             hasura_resource_name=DEFAULT_HASURA_RESOURCE_NAME,
-            port=DEFAULT_HASURA_PORT, 
             hasura_secret=DEFAULT_HASURA_SECRET,
+            hasura_endpoint= 'http://localhost:8080',
             wait_for_services=[]):
-    endpoint = 'http://localhost:{}'.format(port)
     # * Wait for Hasura, then run the console locally
     #  ? fetch admin secret from k8s?
     #  "--admin-secret `kubectl get secret {} -o jsonpath='{{.data.adminSecret}}' | base64 -D`".format(hasura_resource),
     console_options = '--project {} --endpoint {} --admin-secret {}'.format(
         path,
-        endpoint,
+        hasura_endpoint,
         hasura_secret
     )
     config = read_yaml(os.path.join(path, 'config.yaml'), False)
     if not config:
         local('mkdir -p {}'.format(os.path.join(path, '..')))
-        local('hasura init {} --version 3 --endpoint {}'.format(path, endpoint))
+        local('hasura init {} --version 3 --endpoint {}'.format(path, hasura_endpoint))
         # * Bug in the latest hasura cli version - set metadata version.yaml to 3
         # local('echo "version: 3" > {}'.format(
         #     os.path.join(path, 'metadata/version.yaml')))
 
     serve_cmds = [
         'hasura migrate apply {} --all-databases'.format(console_options),
-        'hasura metadata apply {}'.format(console_options),
+        '[ $(wc -c <"{}/metadata/databases/databases.yaml") -ge 4 ] && hasura metadata apply {} || hasura metadata export {}'.format(path, console_options, console_options),
         'hasura console {}'.format(console_options)
     ]
     hasura_resource = '{}-{}'.format(release_name,
                                      hasura_resource_name) if release_name else hasura_resource_name
     console_resource = '{}-hasura-console'.format(
         release_name) if release_name else 'hasura-console'
     local_resource(console_resource,
-                    '\n'.join([wait_for(x) for x in wait_for_services + ['{}/healthz'.format(endpoint)]]),
+                    '\n'.join([wait_for(x) for x in wait_for_services + ['{}/healthz'.format(hasura_endpoint)]]),
                    resource_deps=[hasura_resource],
                    serve_cmd=' && '.join(serve_cmds),
```

Comment:
with the change on line 37, this ends up assembling something like:

migrate && wc && metadata apply || metadata export && console

which, IIUC, will behave as intended, but it might be nice to change this to something like:
`' && '.join(["{{ {}; }}".format(cmd) for cmd in serve_cmds])'` to save readers from thinking about operator precedence